File: test/filesystem/file/display.cpp
#include <ostream>
#include <string>

#include "file.hpp"

std::ostream &operator<<(std::ostream &stream, const File &file) {
    file.display(stream, "", "");
    return stream;
}

void File::display(std::ostream &stream,
        const std::string &leader,
        const std::string &starter) const {

    stream << leader << starter << this->name << '\n';
}

File: source/tree/asset/source/display.go
package source

import "fmt"

func Display(instance *Source, leader string, final bool) {
	starter := ""
	if final {
		starter = "╰─ "
	} else {
		starter = "├─ "
	}

	fmt.Printf("%s%s%s\n", leader, starter, instance.Name)

	for headerName, _ := range instance.Headers {
		newLeader := ""
		if final {
			newLeader = leader + "   "
		} else {
			newLeader = leader + "│  "
		}

		fmt.Printf("%s\u001b[36m→\u001b[0m %s\n", newLeader, headerName)
	}
}
